huaweicloud/services/modelarts/resource_huaweicloud_modelartsv2_service.go

```diff
@@ -0,0 +1,1503 @@
+				Description:      `The minimum number of consecutive detection failures.`,
+			},
+			"check_method": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: suppressV2ServiceUpgradeParamDiffs(),
+				Description:      `The method of the health check.`,
+			},
+			// Optional parameters.
+			"command": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: suppressV2ServiceUpgradeParamDiffs(),
+				Description:      `The commands configuration of the health check.`,
+			},
+			"url": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: suppressV2ServiceUpgradeParamDiffs(),
+				Description:      `The request URL of the health check.`,
+			},
+		},
+	}
+}
+
+func isHistoryServiceVersion(historyVersions map[string]interface{}, targetVersion string) bool {
+	_, isExist := historyVersions[targetVersion]
+	return isExist
+}
+
+func suppressV2ServiceUpgradeParamDiffs() schema.SchemaDiffSuppressFunc {
+	return func(_, _, _ string, d *schema.ResourceData) bool {
+		historyVersions := d.Get("history_versions").(map[string]interface{})
+		targetVersion := d.Get("version").(string)
+		return isHistoryServiceVersion(historyVersions, targetVersion)
+	}
+}
+
+// Only available for the float type of the parameter values compare operation.
+func suppressV2ServiceFloatValuesDiffs() schema.SchemaDiffSuppressFunc {
+	return func(k, _, _ string, d *schema.ResourceData) bool {
+		oldVal, newVal := d.GetChange(k)
+		oldF, oldValIsTypeFloat := oldVal.(float64)
+		newF, newValIsTypeFloat := newVal.(float64)
+		if !oldValIsTypeFloat || !newValIsTypeFloat {
+			return false
+		}
+
+		return utils.EqualFloat(utils.Round(oldF, 2), utils.Round(newF, 2))
+	}
+}
+
+func buildV2ServiceUnitConfigCustomSpec(specConfigs []interface{}) map[string]interface{} {
+	// All sub parameter behaviors are optional.
+	if len(specConfigs) < 1 || specConfigs[0] == nil {
+		return nil
+	}
+
+	specConfig := specConfigs[0]
+	return map[string]interface{}{
+		"gpu":    utils.ValueIgnoreEmpty(utils.PathSearch("gpu", specConfig, nil)),
+		"memory": utils.ValueIgnoreEmpty(utils.PathSearch("memory", specConfig, nil)),
+		"cpu":    utils.ValueIgnoreEmpty(utils.PathSearch("cpu", specConfig, nil)),
+		"ascend": utils.ValueIgnoreEmpty(utils.PathSearch("ascend", specConfig, nil)),
+	}
+}
+
+func buildV2ServiceUnitConfigImage(imageConfigs []interface{}) map[string]interface{} {
+	if len(imageConfigs) < 1 {
+		return nil
+	}
+
+	imageConfig := imageConfigs[0]
+	return map[string]interface{}{
+		"source":   utils.PathSearch("source", imageConfig, "").(string),
+		"swr_path": utils.PathSearch("swr_path", imageConfig, nil),
+		// Only 'IMAGE' source type of image requires this parameter input.
+		"id": utils.ValueIgnoreEmpty(utils.PathSearch("source == 'IMAGE' && id || ''", imageConfig, nil)),
```

Comment:
This conditional filter looks a bit strange, please confirm if there is any problem